mmseg/core/evaluation/metrics.py

```diff
@@ -112,8 +112,6 @@ def total_intersect_and_union(results,
          ndarray: The prediction histogram on all classes.
          ndarray: The ground truth histogram on all classes.
     """
-    num_imgs = len(results)
-    assert len(list(gt_seg_maps)) == num_imgs
```

Comment:
Why remove these assert?

Reply:
`list(gt_seg_maps)` will loop the generator and then `gt_seg_maps` will be empty, https://github.com/open-mmlab/mmsegmentation/blob/4981ff68c2ae05ccfe83e340d528bb4c3c130740/mmseg/core/evaluation/metrics.py#L121 will case error, lead metric to be nan,  I have add unittest in https://github.com/open-mmlab/mmsegmentation/pull/833/commits/bf486901ccfa0addd4153c8cff2f7b2aa4a556d1.

Reply:
There are still missing some lines.
You can view it through `files changed`.

![image](https://user-images.githubusercontent.com/16716329/131873726-628d932e-4d48-4c66-9e60-13070b7fb078.png)

Reply:
If the assert is not remove, https://github.com/open-mmlab/mmsegmentation/blob/4981ff68c2ae05ccfe83e340d528bb4c3c130740/tests/test_data/test_dataset.py#L172 The eval result will be nan